Components/PanoramicDialog.cpp

```diff
@@ -990,98 +1042,57 @@ void
 PanoramicDialog::onNewZoomLevel(float)
 {
   qint64 min, max;
-  qint64 fc =
-        this->ui->waterfall->getCenterFreq()
-        + this->ui->waterfall->getFftCenterFreq();
-  qint64 span = static_cast<qint64>(this->ui->waterfall->getSpanFreq());
-  bool adjLeft = false;
-  bool adjRight = false;
-
-  if (!this->adjustingRange) {
-    this->adjustingRange = true;
-
-    min = fc - span / 2;
-    max = fc + span / 2;
-
-    if (min < this->getMinFreq() && max <= this->getMaxFreq()) {
-      // Too much zooming on the left. Reinject it to the max
-      qint64 extra = static_cast<qint64>(this->getMinFreq()) - min;
-      min += extra;
-      max += extra;
-      adjLeft = adjRight = true;
-    } else if (min >= this->getMinFreq() && max > this->getMaxFreq()) {
-      // Too much zooming on the right. Reinject it to the max
-      qint64 extra = max - static_cast<qint64>(this->getMaxFreq());
-      min -= extra;
-      max -= extra;
-      adjLeft = adjRight = true;
-    }
-
-    if (min < this->getMinFreq()) {
-      min = static_cast<qint64>(this->getMinFreq());
-      adjLeft = true;
-    }
-
-    if (max > this->getMaxFreq()) {
-      max = static_cast<qint64>(this->getMaxFreq());
-      adjRight = true;
-    }
-
-    if (adjLeft && adjRight)
-      this->ui->waterfall->resetHorizontalZoom();
 
-    this->fixedFreqMode = max - min <= this->minBwForZoom * this->getRelBw();
+  getZoomRange(min, max, m_fixedFreqMode);
```

Comment:
Elegant! This kind of changes (isolation of convoluted logic inside a separate method) is always welcome, even if it was not immediately needed.